src/prototype.rs

```diff
@@ -729,20 +781,29 @@ fn parse_line(tree: &mut Tree<Item>, s: &str, mut ix: usize) -> (usize, Option<I
                 ix += 2;
             }
             c @ b'*' | c @b'_' => {
-                tree.append_text(begin_text, ix);
-                let count = 1 + scan_ch_repeat(&s[ix+1..], c);
-                let can_open = ix + count < s.len() && !is_ascii_whitespace(bytes[ix + count]);
-                let can_close = ix > start && !is_ascii_whitespace(bytes[ix - 1]);
-                // TODO: can skip if neither can_open nor can_close
-                for i in 0..count {
-                    tree.append(Item {
-                        start: ix + i,
-                        end: ix + i + 1,
-                        body: ItemBody::Inline(count - i, can_open, can_close),
-                    });
+                // FIXME: it looks like we're mixing byte indices with char indices here.
```

Comment:
I don't think there's a fixme here, there are no char indices here, only byte indices. You're correct there's no panic risk because the delim run is guaranteed to be ASCII.

Reply:
The delim run is ASCII but any characters before it may not be, right? And then we wouldn't reach the delim run from the start of the string.

Reply:
I'm not understanding your concern here, could you spell it out maybe?

Reply:
Yes! Say `s = "ö*"`, then by the time we reach line of the FIXME, the byte counter `ix` will be `2`, and we will panic on `&s[(ix + 1)..]` I think.

Reply:
I'm still not seeing a problem, and I also tested with some Unicode strings. That string is `[0xc3, 0xb6, 0x2a]`, ix is 2 as you say, so ix + 1 is 3 and s[3..] is the empty string. This seems to me as it should be.

Reply:
Aaaah - I completely misunderstood how string indexing works. I thought it was character based! Then the behaviour here is fine indeed. I shall remove the FIXME. Thanks for the clarification!